core/bootstrap.go

```diff
@@ -86,7 +86,7 @@ func bootstrap(ctx context.Context,
 
 	// we can try running dht bootstrap even if we're connected to all bootstrap peers.
 	if len(h.Network().Conns()) > 0 {
-		if err := r.Bootstrap(ctx, numDHTBootstrapQueries); err != nil {
+		if _, err := r.Bootstrap(); err != nil {
```

Comment:
Spitballing: maybe we want to extract the connection supervisor just as we've extracted the reprovider. (We'd still invoke it in the same place.)